src/ZkFold/Base/Protocol/ARK/Plonk.hs

```diff
@@ -114,13 +118,35 @@ newtype PlonkWitnessMap d = PlonkWitnessMap (Map.Map Natural F -> (PolyVec F d,
 
 newtype PlonkWitnessInput = PlonkWitnessInput (Map.Map Natural F)
 -- TODO (Issue #25): make a proper implementation of Show
-instance Show PlonkWitnessInput where
-    show _ = "PlonkWitnessInput"
--- TODO (Issue #25): make a proper implementation of Arbitrary
 instance Arbitrary PlonkWitnessInput where
```

Comment:
We cannot properly define this instance without knowing a circuit. Since we also have `ACandWitness`, I suggest removing this one.